advanced03-AF_XDP/af_xdp_user.c

```diff
@@ -548,10 +548,6 @@ int main(int argc, char **argv)
 		xdp_opts.opts = &opts;
 
 		if (cfg.progname[0] != 0) {
-			xdp_opts.open_filename = cfg.filename;
-			xdp_opts.prog_name = cfg.progname;
-			xdp_opts.opts = &opts;
-
 			prog = xdp_program__create(&xdp_opts);
 		} else {
 			prog = xdp_program__open_file(cfg.filename,
```

Comment:
I think we can go further here. There's no reason not to use `xdp_program__create()` in both cases; and we're not setting any options in the `opts` structure, so there's no reason to include it either. Meaning that this whole section could be collapsed to:
```c
	if (cfg.filename[0] != 0) {
		DECLARE_LIBXDP_OPTS(xdp_program_opts, xdp_opts,
			.open_filename = cfg.filename,
		);
		struct bpf_map *map;

		custom_xsk = true;

		if (cfg.progname[0] != 0)
			xdp_opts.prog_name = cfg.progname;

		prog = xdp_program__create(&xdp_opts);
```

And then just get rid of both the `DECLARE_LIBBPF_OPTS` and `DECLARE_LIBXDP_OPTS` calls at the top of the function.